aptos-move/framework/aptos-stdlib/sources/simple_map.move

```diff
@@ -77,6 +77,38 @@ module aptos_std::simple_map {
         vector::push_back(&mut map.data, Element { key, value });
     }
 
+    public inline fun upsert<Key: store, Value: store>(
+        map: &mut SimpleMap<Key, Value>,
+        key: Key,
+        value: Value,
+        drop: |Key, Value|
+    ) {
+        let len = std::vector::length(&map.data);
+        let i = 0;
+        while (i < len) {
+            let element = vector::borrow(&mut map.data, i);
+            if (&element.key == &key) {
+                break
+            };
+            i = i + 1;
+        };
+        if (i == len) {
+            vector::push_back(&mut map.data, Element { key, value });
+        } else {
+            let Element {key: _k, value: _v} = vector::swap_remove(&mut map.data, i);
+            vector::push_back(&mut map.data, Element { key, value});
+            drop(_k, _v);
+        }
+    }
+
+    public inline fun destroy<Key: store, Value: store>(
+        map: SimpleMap<Key, Value>,
+        d: |Key, Value|
```

Comment:
What do you think about developer experience passing along a drop function? Is it weird at all?

Reply:
Yes, I think both functions may just want `drop` kvs. for undroppable kvs, user should handle them carefully with their own logic since a lambda doesn't help much in that case.

Reply:
I think for destroy it's fine. I wrote two upserts one with drop lambda, one returning an optional